### DISCUSSION TITLE
Python script to list album ID's

### DISCUSSION THREAD

**Discussion by RobWW** [damongolding/immich-kiosk#220] [General]
Thanks for adding the include and exclude album feature.

If it is of use to others, here is a short python script to list album ID's and names from Immich in a form that can be added to the config.yaml file.  Each album produces a line such as:
` -  44a1d764-46fd-4d84-b328-d031f5f5cfba  #  Travel-Provence`

I use a simple grep command to get a subset of albums I am interested in.  For example
`python3 ListAlbums.py | grep -i travel`
gives all the album that have travel in their name.

Rob

```python
import requests
from urllib.parse import urljoin

# Configuration required to access Immich
apiKey = "ZzyYxXWwVvUuTtSsRrQqPpOoa9B8c3D4e5F6g7H8i9JO0"  #  Obtain from Immich (Account Setting ->  Api Keys -> New Api Key)
apiUrl = "http://192.168.1.2:2283/"  #  Immich Url

requests_kwargs = {
    'headers' : {
        'x-api-key': apiKey,
        'Content-Type': 'application/json',
        'Accept': 'application/json'
    }
}

# Fetches assets from the Immich API
def fetchAlbums():
   
    r = requests.get(urljoin(apiUrl, 'api/albums'), **requests_kwargs)
    r.raise_for_status()
       
    albums ={}
    for album in  r.json():
        albums[album['id']] = album['albumName']
    
    return albums
    

albums = fetchAlbums()

for id, name in albums.items():
    print(" -", id, " # ", name)

```

**Comment by damongolding** [comment]
Neat! 

You might want to `join` the  Immich uri and api endpoint. Just to be a bit more fool proof for  newer python users.

**Comment by RobWW** [comment]
To confirm, you are suggesting replacing:
`r = requests.get(apiUrl + 'api/albums',  **requests_kwargs)`

with:
`r = requests.get(''.join([apiUrl, 'api/albums']),  **requests_kwargs)`

to make sure the Url is a string?

**Comment by damongolding** [comment]
I was suggesting:

```Python
from urllib.parse import urljoin
...

r = requests.get(urljoin(apiUrl, 'api/albums'), **requests_kwargs)

```

That way if the user forgets the trailing / off the Immich url it will still work.

**Comment by RobWW** [comment]
Thanks.  That looks good!   I have updated the script in my initial post.
